pkg/model/awsmodel/network.go

```diff
@@ -615,3 +635,104 @@ func (b *NetworkModelBuilder) Build(c *fi.ModelBuilderContext) error {
 
 	return nil
 }
+
+func addAdditionalRoutes(routes []kops.RouteSpec, sbName string, rt *awstasks.RouteTable, lf fi.Lifecycle, c *fi.ModelBuilderContext) error {
+	for _, r := range routes {
+		if strings.HasPrefix(r.Target, "pcx-") {
+			t := &awstasks.Route{
+				Name:                 fi.String("public-" + sbName + "." + r.CIDR),
+				Lifecycle:            lf,
+				CIDR:                 fi.String(r.CIDR),
+				RouteTable:           rt,
+				VPCPeeringConnection: fi.String(r.Target),
+			}
+			c.AddTask(t)
```

Comment:
nit: we could avoid some duplication without sacrificing readability if we define one route task and then conditionally set the appropriate field based on the target
```suggestion
	for _, r := range routes {
		t := &awstasks.Route{
			Name:       fi.String("public-" + sbName + "." + r.CIDR),
			Lifecycle:  lf,
			CIDR:       fi.String(r.CIDR),
			RouteTable: rt,
		}

		if strings.HasPrefix(r.Target, "pcx-") {
			t.VPCPeeringConnection = fi.String(r.Target)
			c.AddTask(t)
```